powersimdata/design/generation/cost_curves.py

```diff
@@ -211,60 +218,62 @@ def build_supply_curve(grid, num_segments, area, gen_type, area_type=None, plot=
         )
 
     # Obtain the desired generator cost and plant information data
-    data = get_supply_data(grid, num_segments)
+    supply_data = get_supply_data(grid, num_segments)
 
     # Check the input supply data
-    check_supply_data(data, num_segments)
+    check_supply_data(supply_data, num_segments)
 
     # Check to make sure the generator type is valid
-    if gen_type not in data["type"].unique():
+    if gen_type not in supply_data["type"].unique():
         raise ValueError(f"{gen_type} is not a valid generation type.")
 
     # Identify the load zones that correspond to the specified area and area_type
     returned_zones = area_to_loadzone(grid.get_grid_model(), area, area_type)
 
     # Trim the DataFrame to only be of the desired area and generation type
-    data = data.loc[data.zone_name.isin(returned_zones)]
-    data = data.loc[data["type"] == gen_type]
+    supply_data = supply_data.loc[supply_data.zone_name.isin(returned_zones)]
+    supply_data = supply_data.loc[supply_data["type"] == gen_type]
 
     # Remove generators that have no capacity (e.g., Maine coal generators)
-    if data["slope1"].isnull().values.any():
-        data.dropna(subset=["slope1"], inplace=True)
+    if supply_data["slope1"].isnull().values.any():
+        supply_data.dropna(subset=["slope1"], inplace=True)
 
     # Check if the area contains generators of the specified type
-    if data.empty:
+    if supply_data.empty:
         return [], []
 
     # Combine the p_diff and slope information for each cost segment
-    df_cols = []
+    supply_df_cols = []
     for i in range(num_segments):
-        df_cols.append(data.loc[:, ("p_diff" + str(i + 1), "slope" + str(i + 1))])
-        df_cols[i].rename(
+        supply_df_cols.append(
+            supply_data.loc[:, ("p_diff" + str(i + 1), "slope" + str(i + 1))]
+        )
+        supply_df_cols[i].rename(
             columns={"p_diff" + str(i + 1): "p_diff", "slope" + str(i + 1): "slope"},
             inplace=True,
         )
-    df = pd.concat(df_cols, axis=0)
+    supply_df = pd.concat(supply_df_cols, axis=0)
 
     # Sort the trimmed DataFrame by slope
-    df = df.sort_values(by="slope")
-    df = df.reset_index(drop=True)
+    supply_df = supply_df.sort_values(by="slope")
+    supply_df = supply_df.reset_index(drop=True)
 
     # Determine the points that comprise the supply curve
-    P = []  # noqa: N806
-    F = []  # noqa: N806
-    p_diff_sum = 0
-    for i in df.index:
-        P.append(p_diff_sum)
-        F.append(df["slope"][i])
-        P.append(df["p_diff"][i] + p_diff_sum)
-        F.append(df["slope"][i])
-        p_diff_sum += df["p_diff"][i]
+    capacity_data = []  # noqa: N806
```

Comment:
I think we can remove all the `noqa` comments on the lines you changed, now that the naming follows pep8.

Reply:
Okay I can remove those. I wasn't sure what the `noqa` comments meant.

Reply:
Done.

Reply:
Thanks!